### DISCUSSION TITLE
No Firebase App '[DEFAULT]' has been created - call firebase.initializeApp() on Android despite correct setup

### DISCUSSION THREAD

**Discussion by DragomirAndrei19** [invertase/react-native-firebase#7462] [Q&A] [answered]
React Native 0.71.3 + Typescript. Project uses ProGuard too, apparently.

I'm trying to integrade push notifications through cloud messaging. I've only tested it on Android.

I did all the steps required. What am I missing?

1) npm install --save @react-native-firebase/app
2) placed my google-services.json file in /android/app/ folder
3) Added `classpath 'com.google.gms:google-services:4.4.0'` at `/android/build.gradle`
4) Added apply plugin: `'com.google.gms.google-services' `at /android/app/build.gradle 
5) npm install @react-native-firebase/messaging
6) npx react-native run-android to rebuild the project
7)` import messaging from '@react-native-firebase/messaging'; `in my App.tsx
8) For testing, I'm adding this function from the documentation
```
async function requestUserPermission() {
  const authStatus = await messaging().requestPermission();
  const enabled =
    authStatus === messaging.AuthorizationStatus.AUTHORIZED ||
    authStatus === messaging.AuthorizationStatus.PROVISIONAL;

  if (enabled) {
    console.log('Authorization status:', authStatus);
  }
}
```
9) I call the function inside an useEffect() in App.tsx

Yet, despite all these steps, I keep getting
```
 Possible Unhandled Promise Rejection (id: 0):
Error: No Firebase App '[DEFAULT]' has been created - call firebase.initializeApp()
```
What am I missing? This is driving me insane.

**Comment by mikehardy** [comment]
Hey there 👋  - there must be some steps after step 9? Like "then I run the pod install / build for whatever platform etc" ? I'm not sure which platform the problem is on so I can't say what those commands should be but if you specify what you do I may be able to help more.

Alternatively you can have a quick read through the comments on my "build a firebase app from scratch, start to finish in one script" script where I demonstrate how to do the integration: https://github.com/mikehardy/rnfbdemo/blob/main/make-demo.sh

**Comment by sander48k** [comment]
Hi. I'm having the same issue, which has also drove me insane, well almost. ;-). My findings so far:

- In my case, the `NATIVE_FIREBASE_APPS` array of the `RNFBAppModule` Native Module seems to be empty on Android. On iOS, this array contains 1 element, which is a reference to the Firebase App on the native side. On Android, this array is filled in the `getConstants` method of the `ReactNativeFirebaseAppModule` (contained in the @react-native-firebase/app package).
- To research the issue further, I have set up 2 basic React Native projects (using the `npx react-native init` command). One project uses RN version 0.71.13 and the other version 0.72.7. To both projects, I have added the latest @react-native-firebase/app package (18.6.1) and added the google services gradle plugin and the google services json file. On the 0.72.7 version, everything works as expected and on the 0.71.13 version the same error is seen.
- The Google Services grade plugin should parse the google services json file and create 1 (or 2?) xml files (see `android/app/build/generated/res/`) that are then used at runtime to initialize the Firebase objects. In the 0.71.13 project, these files are not created while on the 0.72.7 version, these are correctly created.

I suspect that there is some version incompatibility issue between Gradle and the Google Services gradle plugin.

**Reply by mikehardy** [reply]
@sander48k very interesting finding. rn71 certainly worked when I was build-testing while it was out

At the time the google-services plugin version was 4.3.15 - have you tried that one in combination with react-native 0.71.x ?

I've branched my reproducible build-test repo at the commit where rn71 was last used (implying it uses google services plugin 4.3.15, not current / 4.4.0),

I'm running a build test now https://github.com/mikehardy/rnfbdemo/blob/rn71/make-demo.sh but may run out of time today before I can report more results.

It does seem that upgrading to react-native 0.72 is a obvious pathway to success here though :-)

**Reply by mikehardy** [reply, marked as answer]
Reproduced. On rn71 stock templates if you use google-services 4.4.0 it fails on android in exactly the way you describe

If you use rn71 you must use google-services 4.3.15 - it works fine
If you use rn72 you may update to google-services 4.4.0 - it works fine

I'll post a note in the install documentation

Thanks for noticing this!

**Reply by sander48k** [reply]
Thanks Mike! Good to know that version 4.3.15 works with my rn71 project!

**Reply by kundanthe** [reply]
Hi @sander48k @mikehardy 

I am using this configuration and tried both the google-service version but Its not working for me.

    "react-native": "0.72.7",
     classpath 'com.google.gms:google-services:4.3.15'
     distributionUrl=https\://services.gradle.org/distributions/gradle-7.5-bin.zip

**Reply by GouthamJM** [reply]
I am also facing the same issue

**Reply by AjayGol** [reply]
I am also facing the same issue. @mikehardy

**Reply by mikehardy** [reply]
I no longer reproduce this as long as this is true: if you are using gradle < 8, you must not use android google-services gradle plugin greater than 4.3.15

https://github.com/mikehardy/rnfbdemo/blob/rn71/make-demo.sh
https://github.com/mikehardy/rnfbdemo/blob/main/make-demo.sh

Both of those consistently reproduce successful creation of the default firebase app on app startup.

No one has posted a full reproduction of a failure

**Reply by kostas64** [reply]
I had the same issue. In a project with react native 0.71.x it was not working with 'com.google.gms:google-services:4.4.1' (build with no issue just received the header error), but it worked like a charm with 'com.google.gms:google-services:4.3.15'. The misleading for my case was the [docs](https://rnfirebase.io/). 

I attach the reference - in my case my gradle plugin is 7.3.1 so i supposed that it will work also with 4.4.1
```
buildscript {
  dependencies {
    // ... other dependencies
    // NOTE: if you are on react-native 0.71 or below, you must not update
    //       the google-services plugin past version 4.3.15 as it requires gradle >= 7.3.0
    classpath 'com.google.gms:google-services:4.4.1'
    // Add me --- /\
  }
}
```

**Reply by RenanMomesso** [reply]
Thanks so much man. It worked for me!

**Reply by sidceptron** [reply]
Thanks, it helped

**Reply by Ibtixam** [reply]
I am using react native 0.73 version

**Reply by noname0486** [reply]
Thank you very much. react-native: 0.71 works with classpath 'com.google.gms:google-services:4.3.15'

**Comment by AminDannak** [comment]
in my case, another problem was that i had missed the step 4 which @DragomirAndrei19 mentioned: 
"Added apply plugin: 'com.google.gms.google-services' at /android/app/build.gradle"

(it's just one line in the documentation and i had missed it! took me a day to figure that out😑)

anyway I ended up using RN `0.72.6` and `com.google.gms:google-services:4.4.0`.

**Reply by NazimMertBilgi** [reply]
thank you its worked. its my saved day :)

**Reply by mikehardy** [reply]
Should be 4.4.2 now though :-), cheers
https://firebase.google.com/support/release-notes/android (all versions are listed there, our docs should have the versions up to date as well)

**Comment by rp11-coder** [comment]
Hi @mikehardy 
No Firebase App '[DEFAULT]' has been created - call firebase.initializeApp()
RN 0.73.6
gradle version 8
classpath 'com.google.gms:google-services:4.4.1'

**Reply by mikehardy** [reply]
@zocial-rajesh --> does not reproduce: https://github.com/mikehardy/rnfbdemo/blob/main/make-demo.sh

You might examine the output of that from-scratch successful-build reproducer and see if you can find the difference in your gradle files, that should get you going

**Comment by kuldipopenxcell** [comment]
I have same issue in RN latest version:

(NOBRIDGE) ERROR  Error: No Firebase App '[DEFAULT]' has been created - call firebase.initializeApp()

"react-native": "0.74.0"
classpath("com.google.gms:google-services:4.4.1")

**Comment by anujmv** [comment]
Same issue on react native 0.74

(NOBRIDGE) ERROR Error: No Firebase App '[DEFAULT]' has been created - call firebase.initializeApp()

"react-native": "0.74.0"
classpath("com.google.gms:google-services:4.4.1")

**Comment by sashrika** [comment]
You need to configure crashlytics(or any other firebase services which u want) separately after configuring firebase https://rnfirebase.io/crashlytics/usage
There are some additional steps here.

**Comment by anujmv** [comment]
https://github.com/invertase/react-native-firebase/issues/7788#issuecomment-2107941423

the fix for react-native 0.74

**Comment by Mihai-github** [comment]
Does anybody know why I get this error?

Plugin with id 'com.google.gms.google-service' not found

**Reply by kostas64** [reply]
Maybe you forgot to add the plugin in your app/build.gradle?

**Comment by humayune01** [comment]
I'm using Expo and using typescript. In my case plugins were not working so I had to go with [React Native CLI - Android Setup](https://rnfirebase.io/#2-react-native-cli---android-setup) instead of [Expo](https://rnfirebase.io/#expo).

"react-native": "0.74.3"
"expo": "^51.0.21"
"@react-native-firebase/app": "^20.3.0"
"@react-native-firebase/messaging": "^20.3.0"

**Reply by mellah-kais** [reply]
> I'm using Expo and using typescript. In my case plugins were not working so I had to go with [React Native CLI - Android Setup](https://rnfirebase.io/#2-react-native-cli---android-setup) instead of [Expo](https://rnfirebase.io/#expo).

[React Native CLI - Android Setup](https://rnfirebase.io/#2-react-native-cli---android-setup) documentation for expo solved my issue 

adding in app.json :

"plugins": [
      "@react-native-firebase/app",
      "@react-native-firebase/messaging"
      ]
      
solved the "call firebase.InitializeApp()"  issue, 

still didn't get the notification so i added the permission of expo-notifications and it's working now...

**Reply by kdechant** [reply]
Adding the plugins to app.json worked (I'm running RN 0.74.5) but I also had to run `npx expo prebuild --clean` before it would add the `com.google.gms:google-services:4.4.1` line to my build.gradle.

**Comment by AmineKenzup** [comment]
Double-check if `apply plugin: 'com.google.gms.google-services'` is added inside `build.gradle` located inside `android/app` folder

Ref: https://github.com/invertase/react-native-firebase/issues/7788#issuecomment-2188647237

**Comment by gruman** [comment]
I forgot to add my iOS credentials to AppDelegate.m, and that I have it works!

Check out items 2 and 3 here: https://rnfirebase.io/

**Comment by mg-i-ste-mas** [comment]
I resolved the issue by updating the package.json dependencies for Firebase:

```
"dependencies": {
  "@react-native-firebase/app": "^21.3.0",
  "@react-native-firebase/messaging": "^21.3.0"
}
```

I found the latest compatible versions on the npm website.

**Reply by MarcosCunhaDev** [reply]
> I resolved the issue by updating the package.json dependencies for Firebase:
> 
> ```
> "dependencies": {
>   "@react-native-firebase/app": "^21.3.0",
>   "@react-native-firebase/messaging": "^21.3.0"
> }
> ```
> 
> I found the latest compatible versions on the npm website.

This solution worked for me! I had to update all my firebase dependencies and I'm using the following versions:
 - RN: 0.76.1;
 - @react-native-firebase/app: 21.5.0;

**Reply by shuvajitmaitra** [reply]
> "@react-native-firebase/app": "^21.3.0",
>   "@react-native-firebase/messaging": "^21.3.0"

it worked for me

**Reply by alejandro-technology** [reply]
This works for me
    "react-native": "0.76.6",
        "@react-native-firebase/app": "21.3.0",

**Reply by camboYY** [reply]
> 21.3.0

this is working

**Comment by KLichtenauer** [comment]
Tried to fix it with the following setup: 

Dependencies Tried: 
    "react-native": "0.75.4",
    "expo": "^51.0.31",
    "@react-native-firebase/app": "^21.3.0" AND "^21.6.1"
    "@react-native-firebase/messaging": "^21.3.0" AND  "^21.6.1",


Different try:
    "expo": "^52.0.11" with
   "react-native": "0.76.3"

    react-native failed to have compatibility for aws/amplify UI, so didn't get to check if it was working with Firebase.

**Reply by mikehardy** [reply]
@KLichtenauer 
https://stackoverflow.com/help/minimal-reproducible-example
https://stackoverflow.com/help/how-to-ask

starter https://github.com/mikehardy/rnfbdemo/blob/main/make-demo.sh

**Comment by Muhammadmahadsaeed** [comment]
@mikehardy i am facing same issue on android and ios. i have checked the confirguration multiple times and its correct also google-service.json and GoogleService-Info.plist is placed in correct directory. Here is my version of firebase and react native
"@react-native-firebase/app": "21.10.1",
"@react-native-firebase/messaging": "21.10.1",
"react": "18.3.1",
"react-native": "0.76.9",

**Reply by mikehardy** [reply]
So, you are on an old react-native, you likely need google-services gradle plugin 4.3.15, as indicated above @Muhammadmahadsaeed 

What version of that plugin are you using?

**Reply by Muhammadmahadsaeed** [reply]
@mikehardy i am using this classpath 'com.google.gms:google-services:4.4.2' and also tried with 4.3.15 but still facing same issue on android and ios

**Comment by vasylnahuliak** [comment]
I got that error after upgrading React Native from 0.72 to 0.77 version

New Arch: disabled
Hermes: disabled
Firebase: 12.9.3

My fix in `android/build.gradle`

```diff
-        classpath 'com.google.gms:google-services:4.3.10'
+        classpath 'com.google.gms:google-services:4.4.3'
```

I hope it will help someone

https://github.com/invertase/react-native-firebase/issues/2701#issuecomment-3095730621

**Reply by maxedon** [reply]
The solution for me was to revert to the old value for `newArchEnabled=false`, upgrading react-native sets this value to `true`, which causes problems.

**Comment by gautamsamyak** [comment]
Same issue, I am using "@react-native-firebase/app": "^23.3.0"

**Comment by Wajdy-E** [comment]
getting the same error when trying to run on android. using react-native : 0.81.4. using version 23.3.1 for my firebase packages. using version 4.4.3 for google services.

**Reply by thanhlv263** [reply]
if App.tsx or index call firebase, move to another file

**Comment by Beautistar** [comment]
What about for React Native v0.81?

**Reply by cs-manughian** [reply]
For me, after upgrading Expo SDK v54 and RN v0.81.5, I still had stale RN firebase versions, which were v18. I upgraded to RN firebase "@react-native-firebase/app": "23.8.6", and it fixed the error. The expo doctor or expo install --fix didn't find it. Make sure to clear the cache for the next build. :)
